src/figuresClasses.ts

```diff
@@ -1,19 +1,79 @@
 export interface Figure {
+  shape: string;
+  color: string;
+  getArea(): number;
+}
 
+enum Shape {
+  Triangle = 'triangle',
+  Circle = 'circle',
+  Rectangle = 'rectangle',
 }
 
-export class Triangle {
+type Color = 'green' | 'red' | 'blue';
 
-}
+export class Triangle implements Figure {
+  public shape = Shape.Triangle;
+
+  constructor(
+    public color: Color,
+    public a: number,
+    public b: number,
+    public c: number,
+  ) {
+    if (a <= 0 || b <= 0 || c <= 0 || a + b <= c || a + c <= b || b + c <= a) {
+      throw new Error('Input data is incorrect');
+    }
+  }
 
-export class Circle {
+  getArea(): number {
+    const p: number = (this.a + this.b + this.c) / 2;
+    const s: number = +Math.sqrt(p * (p - this.a) * (p - this.b) * (p - this.c))
+      .toFixed(2);
 
+    return s;
+  }
 }
 
-export class Rectangle {
+export class Circle implements Figure {
+  public shape = Shape.Circle;
+
+  constructor(
+    public color: Color,
+    public r: number,
+  ) {
+    if (r <= 0) {
+      throw new Error('Input data is incorrect');
+    }
+  }
+
+  getArea(): number {
+    const s: number = ((this.r ** 2) * Math.PI);
```

Comment:
Use more descriptive variable names. Instead of `r` => `radius`, `s` => `square` and etc. Fix everywhere